src/api/api_client.rs

```diff
@@ -187,38 +197,44 @@ where
 	}
 }
 
-impl<P, Client, Params> Api<P, Client, Params>
+impl<Signer, Client, Params, Runtime> Api<Signer, Client, Params, Runtime>
 where
-	P: Pair,
-	MultiSigner: From<P::Public>,
+	Signer: Pair,
+	MultiSigner: From<Signer::Public>,
 	Client: RpcClient,
-	Params: ExtrinsicParams<Index, Hash>,
+	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
+	Runtime: BalancesConfig,
+	Runtime::Hash: FromHexString,
+	Runtime::Index: From<u32> + Decode,
+	Runtime::Balance: TryFrom<NumberOrHex> + FromStr,
```

Comment:
We could consider adding those traitbounds to our `FrameSystemConfig` trait or `BalancesConfig` trait. Then we can omit them here.

Reply:
True. But I hope to be able to remove most of these once some traits are introduced. Can you bear with that for now?